compiler/src/scala/quoted/runtime/impl/QuotesImpl.scala

```diff
@@ -2681,18 +2645,72 @@ class QuotesImpl private (using val ctx: Context) extends Quotes, QuoteUnpickler
                    |
                    |
                    |The code of the definition of ${t.symbol} is
-                   |${TreeMethods.show(t)}
+                   |${Printer.TreeCode.show(t)}
                    |
                    |which was found in the code
-                   |${TreeMethods.show(tree)}
+                   |${Printer.TreeCode.show(tree)}
                    |
                    |which has the AST representation
-                   |${TreeMethods.showExtractors(tree)}
+                   |${Printer.TreeCases.show(tree)}
                    |
                    |""".stripMargin)
             case _ => traverseChildren(t)
       }.traverse(tree)
 
+    object Printer extends PrinterModule:
+
+      def TreeCode: Printer[Tree] = new Printer[Tree]:
+        def show(tree: Tree): String =
+          SourceCode.showTree(using QuotesImpl.this)(tree)(SyntaxHighlight.plain, fullNames = true)
```

Comment:
It seems the printers can all be `val`s instead of `def`s?

Reply:
`object`/`lazy val` would be better in that case. This is not part of the public API and we will be able to change it later if we need.

Reply:
I will use `lazy val for now`